packages/backend-for-frontend/test/getCatalogEServiceDescriptor.test.ts

```diff
@@ -0,0 +1,328 @@
+    {
+      id: certifiedAttributeId,
+      name: certifiedAttributeName,
+      description: "mockDescription",
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+    {
+      id: declaredAttributeId,
+      name: declaredAttributeName,
+      description: attributeDescription,
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+    {
+      id: verifiedAttributeId,
+      name: verifiedAttributeName,
+      description: attributeDescription,
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+  ]);
+
+  vi.spyOn(delegationService, "getAllDelegations").mockResolvedValue([]);
+
+  vi.spyOn(agreementService, "getLatestAgreement").mockResolvedValue({
+    id: generateId(),
+    eserviceId: eServiceId,
+    descriptorId: mockDescriptorId,
+    consumerId: authData.organizationId,
+  } as unknown as agreementApi.Agreement);
+
+  vi.spyOn(
+    catalogApiConverter,
+    "toBffCatalogDescriptorEService"
+  ).mockReturnValue(catalogDescriptorEService);
+
+  const catalogService = catalogServiceBuilder(
+    mockCatalogProcessClient,
+    mockTenantProcessClient,
+    mockAgreementProcessClient,
+    mockAttributeProcessClient,
+    mockDelegationProcessClient,
+    mockEServiceTemplateProcessClient,
+    fileManager,
+    config
+  );
+
+  it("should retrieve the descriptor successfully", async () => {
+    expect(
+      await catalogService.getCatalogEServiceDescriptor(
+        eServiceId,
+        mockDescriptorId,
+        bffMockContext
+      )
+    ).toEqual(expectedResult);
+
+    expect(attributeService.getAllBulkAttributes).toHaveBeenCalledWith(
+      mockAttributeProcessClient,
+      bffMockContext.headers,
+      ["certifiedAttributeId", "declaredAttributeId", "verifiedAttributeId"]
+    );
+  });
+
+  it("should throw eserviceDescriptorNotFound if descriptorId cannot be found in eservice's descriptors", async () => {
+    vi.spyOn(mockCatalogProcessClient, "getEServiceById").mockResolvedValueOnce(
+      {
+        ...eService,
+        descriptors: [
+          {
+            ...eServiceDescriptor,
+            id: "invalid-descriptorId",
+          },
+        ],
+      }
+    );
+
+    await expect(
+      catalogService.getCatalogEServiceDescriptor(
+        eServiceId,
+        mockDescriptorId,
+        bffMockContext
+      )
+    ).rejects.toThrowError(
+      eserviceDescriptorNotFound(eServiceId, mockDescriptorId)
+    );
+  });
+
+  it("should throw tenant not found", async () => {
+    const mockTenantClientNotFound = {
+      tenant: {
+        getTenant: vi.fn(),
+      },
+    } as unknown as TenantProcessClient;
+
+    const catalogServiceWithEmptyTenant = catalogServiceBuilder(
+      mockCatalogProcessClient,
+      mockTenantClientNotFound,
+      mockAgreementProcessClient,
+      mockAttributeProcessClient,
+      mockDelegationProcessClient,
+      mockEServiceTemplateProcessClient,
+      fileManager,
+      config
```

Comment:
can we refactor this to edit the existing mock, as in the error case, above, to make the two tests more uniform?

(if the issue is that the mock has been overridden already by the previous test, maybe we can handle that by setting the mock for the OK case with a `beforeEach`)